color/src/cache_key.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2024 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Hashing and other caching utilities for Color types.
+//!
+//! In this crate, colors are implemented using `f32`.
+//! This means that color types aren't `Hash` and `Eq` for good reasons:
+//!
+//! - Equality on these types is not reflexive (consider [NaN](f32::NAN)).
+//! - Certain values have two representations (`-0` and `+0` are both zero).
+//!
+//! However, it is still useful to create caches which key off these values.
+//! These are caches which don't have any semantic meaning, but instead
+//! are used to avoid redundant calculations or storage.
+//!
+//! Color supports creating these caches by using [`CacheKey<T>`] as the key in
+//! your cache.
+//! `T` is the key type (i.e. a color) which you want to use as the key.
```

Comment:
maybe this "i.e." should be "e.g." as we also support slices, and users can themselves implement `BitEq`, `BitHash`.

Reply:
This isn't really intended to be used for non-colour cases (hence the impl only being for f32 and not all the other numeric types). Like yes, you can implement this yourself, but we are being careful about that.